apps/meteor/app/gitlab/client/lib.ts

```diff
@@ -19,26 +18,24 @@ const config: OauthConfig = {
 
 const Gitlab = new CustomOAuth('gitlab', config);
 
-Meteor.startup(() => {
-	Tracker.autorun(() => {
-		let anyChange = false;
-		if (settings.get('API_Gitlab_URL')) {
-			config.serverURL = settings.get('API_Gitlab_URL').trim().replace(/\/*$/, '');
-			anyChange = true;
+export const useGitLab = () => {
+	const gitlabApiUrl = useSetting('API_Gitlab_URL') as string;
+	const gitlabIdentiry = useSetting('Accounts_OAuth_Gitlab_identity_path') as string;
+	const mergeUsers = useSetting('Accounts_OAuth_Gitlab_merge_users') as boolean;
+
+	useEffect(() => {
+		if (gitlabApiUrl) {
+			config.serverURL = gitlabApiUrl.trim().replace(/\/*$/, '');
 		}
 
-		if (settings.get('Accounts_OAuth_Gitlab_identity_path')) {
-			config.identityPath = settings.get('Accounts_OAuth_Gitlab_identity_path').trim() || config.identityPath;
-			anyChange = true;
+		if (gitlabIdentiry) {
+			config.identityPath = gitlabIdentiry.trim() || config.identityPath;
 		}
 
-		if (settings.get('Accounts_OAuth_Gitlab_merge_users')) {
+		if (mergeUsers) {
 			config.mergeUsers = true;
-			anyChange = true;
 		}
 
-		if (anyChange) {
```

Comment:
we won't check if theres change in order to configure it anymore?

Reply:
This seemed like a workaround for reactivity, the new `useEffect` should configure it again when any of the dependencies change

Reply:
oh yeah, makes sense srry didn't realized!